src/editor.rs

```diff
@@ -267,32 +267,21 @@ fn parse_command(keymap: &KeyMap, command: &str) -> Option<Action> {
     let mut command_char_iter = command.chars();
 
     // Consume the first char of the command
-    let c = command_char_iter.next()?;
-
-    match keymap.get(&c) {
-        // "q" quits Sapling
-        Some(Command::Quit) => Some(Action::Quit),
-        Some(Command::InsertChild) => {
-            // Consume the second char of the iterator
-            command_char_iter.next().map(Action::InsertChild)
+    return if let Some(command) = command_char_iter.next().and_then(|c| keymap.get(&c)) {
```

Comment:
The unit tests are failing because the 'refactoring' has removed the possibility for the `parse_command` to return `Some(Action::Undefined)` if the command is invalid.  This is happening because `and_then` does not differentiate between `keymap.get(&c)` and `command_char_iter.next()` returning `None`, which is problematic because they correspond to the command being invalid and incomplete respectively and so should result in two different return types.  So therefore two nested `if let`s are required so that these cases are differentiated.

Also since we are bikeshedding a bit on the command parsing code, we can make this even more idiomatic by replacing the return type `Option<Action>` with an custom Option-like enum with two versions of `None`:

```rust
enum ParseResult<T> {
    Ok(T),
    Invalid,
    Incomplete
}
```
and removing `Action::Undefined` (this will cause some refactoring elsewhere in the code, but I think we'll only be removing some `unreachable!()`s or janky error code).